### PR TITLE
initial commit for redis keyval impl. (SPOPT-811)

### PR DESCRIPTION
**Pull request by junmipan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +22,7 @@ define test_cover_only
 	@echo "# running unit tests with coverage analysis"
 	@go test -covermode=count -coverprofile=${COVER_DIR}coverage_unit1.out ./logging/logrus
 	@go test -covermode=count -coverprofile=${COVER_DIR}coverage_unit2.out ./db/keyval/etcdv3
+	@go test -covermode=count -coverprofile=${COVER_DIR}coverage_unit2.out ./db/keyval/redis
```
Comment (lukasmacko) [inline Makefile]: pleasue use unique filename such as `${COVER_DIR}coverage_unit7.out` and add the file into the list on the line 32

File: db/keyval/redis/bytes_broker_impl.go [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+
+	"bytes"
+
+	"fmt"
+	"github.com/garyburd/redigo/redis"
+)
+
+// BytesConnectionRedis allows to store, read and watch values from Redis.
+type BytesConnectionRedis struct {
+	pool *redis.Pool
```
Comment (jozef-slezak) [inline db/keyval/redis/bytes_broker_impl.go]: Could you please use here interface to be able to replace normal pool with cluster enabled pool?

File: db/keyval/redis/bytes_broker_impl.go [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+// bytesKeyVal represents a single key-value pair
+type bytesKeyVal struct {
+	key   string
+	value []byte
+}
+
+var log logging.Logger
+
+func init() {
+	log = logrus.StandardLogger()
```
Comment (jozef-slezak) [inline db/keyval/redis/bytes_broker_impl.go]: I would like to keep this code dependent only on logging interfaces (not it's implementation). Lukas needs to improve logging package...

File: db/keyval/redis/bytes_broker_impl.go [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+	log.Debugf("ListValues(%s)", match)
+
+	conn := db.pool.Get()
+	defer conn.Close()
+	keys, err := db.listKeys(conn, match)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO NICE-to-HAVE optmize with Redis Cursor - see Redis SCAN comman
```
Comment (jozef-slezak) [inline db/keyval/redis/bytes_broker_impl.go]: Could you please use Redis Cursor in iterator?

**Pull request merged**